src/_pytest/capture.py

```diff
@@ -632,20 +641,23 @@ def snap(self):
 
 
 class SysCapture:
-
     EMPTY_BUFFER = str()
     _state = None
 
     def __init__(self, fd, tmpfile=None):
         name = patchsysdict[fd]
         self._old = getattr(sys, name)
         self.name = name
+        self.fd = fd
         if tmpfile is None:
             if name == "stdin":
                 tmpfile = DontReadFromInput()
             else:
-                tmpfile = CaptureIO()
+                tmpfile = self._get_writer_class()
         self.tmpfile = tmpfile
+        
+    def _get_writer_class(self):
+        return CaptureIO()
```

Comment:
Instead of this indirection it should get passed in via `tmpfile` really I think.

Reply:
I could do that in the `MultiCapture` class, but that would require additional conditions that will complicate the code. I could also create a new class that inherits `MultiCapture`, but it would have to basically duplicate its `__init__()` which could become a maintenance problem in the future. None of those seems reasonable to me. Do you have any other suggestions for that?